protocol/migrations/sqlite/1678292329_add_collapsed_categories.up.sql

```diff
@@ -0,0 +1,6 @@
+CREATE TABLE collapsed_community_categories (
+  community_id VARCHAR NOT NULL,
+  category_id VARCHAR NOT NULL,
+  collapsed INTEGER NOT NULL,
+  UNIQUE(community_id, category_id) ON CONFLICT REPLACE
+);
```

Comment:
Is the intent of this implementation to prevent pulling message data about categories that the user has collapsed?

Is this the reason why there is a separate table and not just a `collapsed` bool on the categories?

Reply:
The intent is just to record which category in a community the user has collapsed https://github.com/status-im/status-mobile/issues/15290

It's just a UI thing, we show them as closed (i.e not showing the list of chats within a category), or open (we list the chats in this category), similarly to discord, but other than that, it has no impact.

Regarding storage, categories are stored with the community description, and the community description is a field that is set/overriden by the admin of the community (it should not contain any user-set data), so I think it's best to store this separately, but not sure if I addressed your concern, did I understand the question correctly?

Reply:
Concern addressed.